src/store_client.cc

```diff
@@ -990,7 +990,7 @@ CheckQuickAbortIsReasonable(StoreEntry * entry)
         return false;
     }
 
-    if ((curlen / (expectlen / 100)) > (Config.quickAbort.pct)) {
+    if (curlen > expectlen*(Config.quickAbort.pct/100.0)) {
```

Comment:
If this change eliminates the need for the previous `if` statement (the one that says "avoid FPE"), then please also remove that `if` statement (and associated XXX and TODO comments). Otherwise, please discuss why that `if` statement should be preserved after making division by zero seemingly impossible.

If you do remove that `if` statement, please adjust PR description to briefly document the effect of that removal (e.g., I would expect a slight change in behavior for some setting of quick_abort_pct and possibly other quick_abort_... configuration options). When writing that comment, think as a sysadmin configuring Squid or reviewing their existing configuration (rather than as a Squid developer).

N.B. Here, FPE is spelled like "Floating Point Exception" but really stands for "division by zero". For more information see, for example, this [answer](https://stackoverflow.com/a/46380011) to a mostly unrelated question that says "The Single Unix Specification [defines SIGFPE](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/signal.h.html) as "Erroneous arithmetic operation". It's confusingly named after floating point, but in a normal system with the FPU in its default state, only integer math will raise it. On x86, only integer division".

P.S. The "avoid FPE" protection was added in 1998 commit 4efca7da.

Reply:
That FPE check is no longer required, we won't face a situation of Divison By Zero in this expression. 
I have removed that `if` statement.